### ISSUE TITLE
SortSam fatal error

### ISSUE THREAD

**Issue by jhl667** [broadinstitute/picard#927]
### Affected tool(s)
_SortSam_

### Affected version(s)
- [ ] Latest public release version [2.12.1]
- [ ] I know this was also occurring in 2.7.1, but is not an issue using v1 tools.

### Description 
Core dump occurs when processing some data files, haven't determined the characteristics of the errant samples but some do appear to predictably error.   Out of the latest batch of 30 or so data sets, 3 of them failed with this error:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fd6a652ee8b, pid=64067, tid=0x00007fcb474f9700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_144-b01) (build 1.8.0_144-b01)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.144-b01 mixed mode linux-amd64 )
# Problematic frame:
# V  [libjvm.so+0x64be8b]  InstanceKlass::oop_follow_contents(ParCompactionManager*, oopDesc*)+0x16b
```
Full log is available on request.

#### Steps to reproduce
This was run in a local Galaxy instance, but you can see the general idea.
`_JAVA_OPTIONS=${_JAVA_OPTIONS:-'-Xmx45g -Xms16g'} && export _JAVA_OPTIONS && ln -f -s '/home/exacloud/clinical/Galaxy/database/files/000/373/dataset_373604.dat' 'BWA-MEM_on_data_21__mapped_reads' && picard SortSam INPUT='BWA-MEM_on_data_21__mapped_reads' OUTPUT='/home/exacloud/clinical/Galaxy/database/job_working_directory/000/198/198564/galaxy_dataset_494450.dat' SORT_ORDER="coordinate" QUIET=true VERBOSITY=ERROR VALIDATION_STRINGENCY=LENIENT MAX_RECORDS_IN_RAM=11250000`

**Comment by yfarjoun** [comment]
1. does your input file validate?
2. could this be a java bug? did you try on older version of java (I'm on 1.8.0_77)
3. anything else special about your file (is it big? does it have long reads? it is unaligned?)

**Comment by yfarjoun** [comment]
oh, run again without 
```
QUIET=true VERBOSITY=ERROR VALIDATION_STRINGENCY=LENIENT
```
  please and see if there's something informative

**Comment by tfenne** [comment]
I think that this has to be either a) a JVM bug or b) a bug in a native library.  I think the only native library in use is snappy, so it might be worth disabling snappy as well to see if that's the issue.  To do that you can try adding `-Dsnappy.disable=true` directly after `java` in your command line.

**Comment by jhl667** [comment]
@yfarjoun 
1) I do have a problem @PG read group in the header, but all the previous samples I've run have the exact same format, and I've never come across these errors.  Setting the stringency to "LENIENT" has fixed this is the past, but perhaps this is something I need to go back and fix permanently.
2) I have also used an older version, r65 I believe, and I seem the same behavior.  I actually upgraded to r144 to check and see if it were a known bug.
3) Honestly, I don't think there is anything different with the file.  We are working with small targeted panels, and these have been prepared in an identical manner as the rest of the samples that did not fail.  That being said, it's always possible I'm overlooking something...

Nothing informative running the command without those options, other than the validation error I mentioned above.

@tfenne I added this to the command line I referenced above, and the error still occurs.

This appears to be a JVM bug, though I can't say this is technically true without understanding why it is happening or whether the behavior is intentional.  I am able to run these commands without fail using any combination of parameters I listed.  The error only occurs when I set _JAVA_OPTIONS before the command.  This appears to be a problem in a small number of samples that I process.  We've run 100's, and only seen this behavior 3 times.  _JAVA_OPTIONS is probably not even necessary for us any longer, so I'll just tear it out until I can get a little more info.

Thanks for the suggestions/help!
